tests/unit/test_network_connections.py

```diff
@@ -445,6 +448,57 @@ def test_validate_list(self):
         )
         self.assertEqual([1, 5], v.validate(["1", 5]))
         self.assertValidationError(v, [1, "s"])
+        # Assert that `None` is normalized into `[]`` in ArgvalidatorList()
+        self.assertEqual(v.validate(None), [])
+
+        v2 = network_lsr.argument_validator.ArgValidatorList(
+            "list2",
+            nested=network_lsr.argument_validator.ArgValidatorStr(
+                "list2[?]", allow_empty=True
+            ),
+        )
+        self.assertEqual(
+            v2.validate(["pci-0001:00:00.0", ""]), ["pci-0001:00:00.0", ""]
+        )
+        self.assertRaisesRegexp(
+            ValidationError,
+            "must be a string but is 'None'",
+            v2.validate,
+            ["pci-0001:00:00.0", None],
+        )
+        v3 = network_lsr.argument_validator.ArgValidatorList(
```

Comment:
These should be different test methods for each test case (see the tests that were added later).

Reply:
I splitted the test cases into separate functions.